### PR TITLE
stats: add metrics for stats

### PR DESCRIPTION
**Pull request by lamxTyler**
Add metrics for stats inaccuracy rate.

### CODE REVIEW

File: statistics/update.go [hunk 1]
```diff
@@ -92,6 +94,16 @@ func (s *SessionStatsCollector) StoreQueryFeedback(feedback interface{}) {
...
 	}
+
+	var rate float64
+	if q.actual == 0 {
+		rate = 100
+	} else {
+		rate = math.Abs(float64(q.expected-q.actual)/float64(q.actual)) * 100
+		rate = math.Min(100, rate)
+	}
+	metrics.StatsErrorRate.Observe(rate)
```
Comment (shenli) [inline statistics/update.go]: I guess this is `inaccuracy`.

File: statistics/update.go [hunk 2]
```diff
@@ -92,6 +94,16 @@ func (s *SessionStatsCollector) StoreQueryFeedback(feedback interface{}) {
 	if q.histVersion == 0 || q.actual < 0 || !q.valid {
 		return
 	}
+
+	var rate float64
+	if q.actual == 0 {
+		rate = 100
+	} else {
+		rate = math.Abs(float64(q.expected-q.actual)/float64(q.actual)) * 100
+		rate = math.Min(100, rate)
```
Comment (coocood) [inline statistics/update.go]: So when expected is way larger than actual, the rate is 100.

Comment (lamxTyler) [reply statistics/update.go]: Yes.

File: statistics/update.go [hunk 3]
```diff
@@ -92,6 +94,20 @@ func (s *SessionStatsCollector) StoreQueryFeedback(feedback interface{}) {
...
+	var rate float64
+	if q.actual == 0 {
+		if q.expected == 0 {
+			rate = 0
+		} else {
+			rate = 100
+		}
+	} else {
+		rate = math.Abs(float64(q.expected-q.actual)/float64(q.actual)) * 100
+		rate = math.Min(100, rate)
```
Comment (coocood) [inline statistics/update.go]: I think 100 is too small, it's acceptable in many cases that actual is 2x larger than expected.

File: statistics/update.go [hunk 4]
```diff
@@ -92,6 +94,20 @@ func (s *SessionStatsCollector) StoreQueryFeedback(feedback interface{}) {
...
+
+	var rate float64
+	if q.actual == 0 {
+		if q.expected == 0 {
+			rate = 0
+		} else {
+			rate = 100
+		}
+	} else {
+		rate = math.Abs(float64(q.expected-q.actual)/float64(q.actual)) * 100
```
Comment (coocood) [inline statistics/update.go]: Can we remove the `* 100`?

File: statistics/update.go (hunk 3 above)
Comment (lamxTyler) [reply statistics/update.go]: But knowing that it is 2x larger than actual does not give more information than 1x larger?

File: statistics/update.go (hunk 4 above)
Comment (lamxTyler) [reply statistics/update.go]: We can, but why?

**Comment by shenli** [comment]
LGTM

**Pull request merged**